File: backend/src/main/java/edu/cit/cabasag/barberconnect/dto/response/AuthResponse.java
package edu.cit.cabasag.barberconnect.dto.response;

import edu.cit.cabasag.barberconnect.model.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AuthResponse {
    
    private String firebaseUid;
    private String firstName;
    private String lastName;
    private String email;
    private String phoneNumber;
    private User.UserRole role;
    private boolean isActive;
    private String profileImageUrl;
    private String token; // JWT token
    private BarberProfileResponse barberProfile;
    
    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class BarberProfileResponse {
        private String id;
        private String bio;
        private Integer yearsExperience;
        private String rating;
        private Integer totalReviews;
        private String profileImageUrl;
        private String gcashNumber;
        private Boolean isAvailable;
    }
}

File: mobile/app/src/main/java/edu/cit/cabasag/barberconnect/model/AuthResponse.kt
package edu.cit.cabasag.barberconnect.model

data class AuthResponse(
    val firebaseUid: String? = null,
    val user_id: String? = null,
    val firstName: String? = null,
    val lastName: String? = null,
    val email: String? = null,
    val phoneNumber: String? = null,
    val role: String? = null,
    val isActive: Boolean? = null,
    val profileImageUrl: String? = null,
    val token: String? = null,
    val barberProfile: BarberProfileResponse? = null
) {
    data class BarberProfileResponse(
        val id: String? = null,
        val firstName: String? = null,
        val lastName: String? = null,
        val bio: String? = null,
        val yearsExperience: Int? = null,
        val rating: String? = null,
        val totalReviews: Int? = null,
        val profileImageUrl: String? = null,
        val gcashNumber: String? = null,
        val isAvailable: Boolean? = null
    )

    val resolvedUserId: String?
        get() = firebaseUid ?: user_id

    /** Convenience: "John D." display name */
    fun displayName(): String {
        val first = firstName?.trim() ?: ""
        val last = lastName?.trim()?.firstOrNull()?.let { "$it." } ?: ""
        return "$first $last".trim()
    }

    /** Initials for the avatar circle (e.g. "JD") */
    fun initials(): String {
        val f = firstName?.firstOrNull()?.uppercaseChar() ?: '?'
        val l = lastName?.firstOrNull()?.uppercaseChar() ?: '?'
        return "$f$l"
    }
}
